### PR TITLE
Fix console show log

### LINKED ISSUE #12795: `lxc console --show-log` gives confusing output
`lxc console --show-log` only returns new log messages that were generated after the command was last run. However, if there are no new messages, the command returns `Error: open : no such file or directory` instead of a message that indicates that there are no new messages.

```
ubuntu@ru-fu01:~$ lxc console present-mollusk --show-log

Console log:

systemd 249.11-0ubuntu3.11 running in system mode (+PAM +AUDIT +SELINUX +APPARMOR +IMA +SMACK +SECCOMP +GCRYPT +GNUTLS +OPENSSL +ACL +BLKID +CURL +ELFUTILS +FIDO2 +IDN2 -IDN +IPTC +KMOD +LIBCRYPTSETUP +LIBFDISK +PCRE2 -PWQUALITY -P11KIT -QRENCODE +BZIP2 +LZ4 +XZ +ZLIB +ZSTD -XKBCOMMON +UTMP +SYSVINIT default-hierarchy=unified)
[...]
Ubuntu 22.04.3 LTS present-mollusk console

present-mollusk login: 
ubuntu@ru-fu01:~$ lxc console present-mollusk --show-log
Error: open : no such file or directory
```

When running `lxc console --show-log` on a VM, the error message is `Error: Instance is not container type`. Neither the help nor the documentation say that the command is only for containers. Should it work for both VMs and containers, or do we need to update the documentation?

```
ubuntu@ru-fu01:~$ lxc console my-vm --show-log
Error: Instance is not container type
```

Finally, when running the corresponding API query, the output is the same if there are no new log messages, but if there are, the output is even more confusing:

```
ubuntu@ru-fu01:~$ lxc query --request GET /1.0/instances/present-mollusk/console
Error: open : no such file or directory
ubuntu@ru-fu01:~$ lxc restart present-mollusk 
ubuntu@ru-fu01:~$ lxc query --request GET /1.0/instances/present-mollusk/console
Error: invalid character 's' looking for beginning of value
ubuntu@ru-fu01:~$ lxc query --request GET /1.0/instances/present-mollusk/console
Error: invalid character 'O' looking for beginning of value
ubuntu@ru-fu01:~$ lxc query --request GET /1.0/instances/present-mollusk/console
Error: invalid character 'U' looking for beginning of value
ubuntu@ru-fu01:~$ lxc query --request GET /1.0/instances/present-mollusk/console
Error: open : no such file or directory
```

### PR DESCRIPTION
**Pull request by MggMuggins**
Fixes #12795 

Skipping the improvements for `lxc query` for now, may take a look later.

### CODE REVIEW

**Review by tomponline — COMMENTED** [review]
So this doesnt work for VMs?

**Comment by MggMuggins** [comment]
Yes, it's currently [disallowed](https://github.com/canonical/lxd/blob/main/lxd/instance_console.go#L599).

**Comment by MggMuggins** [comment]
It's already indicated [here](https://documentation.ubuntu.com/lxd/en/latest/howto/instances_troubleshoot/#how-to-troubleshoot-failing-instances) and [here](https://documentation.ubuntu.com/lxd/en/latest/howto/instances_console/#how-to-access-the-console); the only other reference is the lxc cli help.

**Comment by tomponline** [comment]
thanks. Looks like this needs a rebase

**Review by tomponline — APPROVED** [review]

**Comment by MggMuggins** [comment]
Pushed a linter fix

**Review by tomponline — APPROVED** [review]

File: lxc/console.go [hunk 1]
```diff
@@ -123,7 +123,7 @@ func (c *cmdConsole) run(cmd *cobra.Command, args []string) error {
 	// Show the current log if requested
 	if c.flagShowLog {
 		if c.flagType != "console" {
-			return fmt.Errorf(i18n.G("The --show-log flag is only supported for by 'console' output type"))
+			return fmt.Errorf("%s", i18n.G("The --show-log flag is only supported for by 'console' output type"))
```
Comment (tomponline) [inline lxc/console.go:126]: We can just use https://pkg.go.dev/fmt#Sprint

**Review by tomponline — CHANGES_REQUESTED** [review]

File: lxc/console.go (hunk 1 above)
Comment (tomponline) [reply lxc/console.go:126]: Ah not with error pkg

**Pull request merged**